### PR TITLE
[FEATURE] Cosmos | Wire up ConnectionPolicy, DirectConnectionConfig, and RntbdTransportClient.Options

### PR DESCRIPTION
**Pull request by David-Noble-at-work**
In addition default `RntbdTransportClient.Options` are now obtained from `DirectConnectionConfig.defaultConfig`, the definitive source for all default public-facing Direct TCP options.

#### Also

- `AsyncBenchmark` now correctly reports the number of pre-created documents (not the number of operations).
- Removed dead code from `RntbdClientChannelPool`.
- Replaced use of deprecated Jackson method in `RntbdObjectMapper`.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/DirectConnectionConfig.java [hunk 1]
```diff
@@ -17,24 +17,27 @@
...
     private static final Duration DEFAULT_IDLE_ENDPOINT_TIMEOUT = Duration.ofSeconds(70L);
     private static final Duration DEFAULT_CONNECTION_TIMEOUT = Duration.ofSeconds(60L);
+    private static final Duration DEFAULT_REQUEST_TIMEOUT = Duration.ofSeconds(60L);
     private static final int DEFAULT_MAX_CONNECTIONS_PER_ENDPOINT = 30;
     private static final int DEFAULT_MAX_REQUESTS_PER_CONNECTION = 10;
 
     private Duration connectionTimeout;
     private Duration idleConnectionTimeout;
     private Duration idleEndpointTimeout;
+    private Duration requestTimeout;
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/DirectConnectionConfig.java:27]: Please add this new field to `toString()` API.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ConnectionPolicy.java [hunk 2]
```diff
@@ -476,10 +458,9 @@ public String toString() {
             ", inetSocketProxyAddress=" + inetSocketProxyAddress +
             ", readRequestsFallbackEnabled=" + readRequestsFallbackEnabled +
             ", connectionTimeout=" + connectionTimeout +
-            ", idleChannelTimeout=" + idleChannelTimeout +
             ", idleEndpointTimeout=" + idleEndpointTimeout +
-            ", maxChannelsPerEndpoint=" + maxChannelsPerEndpoint +
-            ", maxRequestsPerChannel=" + maxRequestsPerChannel +
+            ", maxChannelsPerEndpoint=" + maxConnectionsPerEndpoint +
+            ", maxRequestsPerChannel=" + maxRequestsPerConnection +
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ConnectionPolicy.java:463]: We should also rename the string wordings -> `maxChannelsPerEndpoint` -> `maxConnectionsPerEndpoint`
and `maxRequestsPerChannel` -> `maxRequestsPerConnection`

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/DirectConnectionConfig.java [hunk 3]
```diff
@@ -190,6 +193,32 @@ public DirectConnectionConfig setMaxRequestsPerConnection(int maxRequestsPerConn
...
 
+    /**
+     * Gets the request timeout interval
+     * This represents the timeout interval for requests
+     *
+     * Default value is 60 seconds
+     *
+     * @return the request timeout interval
+     */
+    public Duration getRequestTimeout() {
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/DirectConnectionConfig.java:204]: Is this the end to end (including the consistency stack) timeout?

Comment (David-Noble-at-work) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/DirectConnectionConfig.java:204]: yes.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/DirectConnectionConfig.java (hunk 1 above)
Comment (David-Noble-at-work) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/DirectConnectionConfig.java:27]: good catch, thanks

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ConnectionPolicy.java (hunk 2 above)
Comment (David-Noble-at-work) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ConnectionPolicy.java]: another good catch. change made on my branch. to be committed with this PR.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/DirectConnectionConfig.java (hunk 3 above)
Comment (moderakh) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/DirectConnectionConfig.java:204]: @kirankumarkolli 
both GWConnectionConfig and DirectConnectionConfig have requestTimeout.
and `DirectConnectionConfig#requestTimeout` represents the end to end latency (including consistency stack).

I wonder if we should keep these at the ConnectionConfig level or if we should move it up to the CosmosClientBuilder. How .Net is doing this?

Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/DirectConnectionConfig.java:204]: @moderakh  - Let's merge this PR as it is, and we will have a follow up PR which will handle these changes - https://github.com/Azure/azure-sdk-for-java/issues/11531

File: sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/AsyncBenchmark.java [hunk 4]
```diff
@@ -131,7 +131,7 @@
         if (configuration.getOperationType() != Configuration.Operation.WriteLatency
                 && configuration.getOperationType() != Configuration.Operation.WriteThroughput
                 && configuration.getOperationType() != Configuration.Operation.ReadMyWrites) {
-            logger.info("PRE-populating {} documents ....", cfg.getNumberOfOperations());
+            logger.info("PRE-populating {} documents ....", cfg.getNumberOfPreCreatedDocuments());
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/AsyncBenchmark.java:134]: good catch. thank you

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/DirectConnectionConfig.java (hunk 3 above)
Comment (moderakh) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/DirectConnectionConfig.java:204]: agreed. closing this as resolved.

**Comment by kushagraThapar** [comment]
CI shows 3 tests failures : 

`validateSessionTokenAsync `
`insertWithUniqueIndex`
`queryDocumentsWithCompositeContinuationTokens`


Common cause - `timeout` : https://dev.azure.com/azure-sdk/internal/_build/results?buildId=408257&view=ms.vss-test-web.build-test-results-tab&runId=11233162&resultId=100016&paneView=debug

**Pull request merged**